src/components/map.tsx

```diff
@@ -0,0 +1,60 @@
+import {useRef, useEffect} from 'react';
+import leaflet from 'leaflet';
+import useMap from '../hooks/use-map.tsx';
+import {Location, OfferType} from '../types.ts';
+import {URL_MARKER_DEFAULT, URL_MARKER_CURRENT} from '../const/const.ts';
+import 'leaflet/dist/leaflet.css';
+import { ICON_WIDTH, ICON_HEIGHT, ANCHOR_X, ANCHOR_Y } from '../const/const.ts';
+
+type MapProps = {
+  city: Location;
+  points: OfferType[];
+  selectedPoint: OfferType | undefined;
+};
+
+const defaultCustomIcon = leaflet.icon({
+  iconUrl: URL_MARKER_DEFAULT,
+  iconSize: [ICON_WIDTH, ICON_HEIGHT],
+  iconAnchor: [ANCHOR_X, ANCHOR_Y]
+});
+
+const currentCustomIcon = leaflet.icon({
+  iconUrl: URL_MARKER_CURRENT,
+  iconSize: [ICON_WIDTH, ICON_HEIGHT],
+  iconAnchor: [ANCHOR_X, ANCHOR_Y]
+});
```

Comment:
Всё ещё много копипасты:

```suggestion
const commonIconSettings = {
  iconSize: [ICON_WIDTH, ICON_HEIGHT],
  iconAnchor: [ANCHOR_X, ANCHOR_Y]
}

const defaultCustomIcon = leaflet.icon({
  ...commonIconSettings,
  iconUrl: URL_MARKER_DEFAULT,
});

const currentCustomIcon = leaflet.icon({
  ...commonIconSettings,
  iconUrl: URL_MARKER_CURRENT,
});
```